### ISSUE TITLE
Using AdaGrad and AdaDelta with SVM and Liner Regression in Spark/Scala

### ISSUE THREAD

**Issue by juan211189** [dmlc/mxnet#4674]
I want to run SVM and Linear Regression in some datasets (using Spark/Scala). However, those methods in Spark only run using Stochastic Gradient Descent with no adaptative learning rates. I want to test them using AdaGrad and AdaDelta, so I would like to know if I there is a way to use the mxnet implementations of these optimizers along with the SVM and Linear Regression methods implemented in Spark (with scala). For instance, I know in Spark it is possible to set the optimizer to be used by SVM, so can I just set the optimizer and use one from mxnet library? if the answer is yes, how should I instantiate the AdaGrad and AdaDelta optimizers?

**Comment by Ldpe2G** [comment]
@javelinjs

**Comment by sethah** [comment]
This is probably a question for Spark itself, not MXNet. The SVM you are referring to is in the old MLlib package, which is deprecated. The optimizer on it can be changed yes, but if you look at the type signatures it needs to extend the `Optimizer` class defined in MLlib. So you'd have to wrap the MXNet optimizers in an `Optimizer` class. I doubt that would be worth your time. SGD on Spark does not work well. The new ML package has a `LinearSVC` which uses quasi-newton methods which will be better on Spark.

Anyway, I think this issue should be closed.
